Wrappers/Python/cil/framework/data_container.py

```diff
@@ -0,0 +1,1548 @@
+
+
+    def apply_circular_mask(self, radius=0.99, in_place=True):
+        """
+
+        Apply a circular mask to the horizontal_x and horizontal_y slices. Values outside this mask will be set to zero.
+
+        This will most commonly be used to mask edge artefacts from standard CT reconstructions with FBP.
+
+        Parameters
+        ----------
+        radius : float, default 0.99
+            radius of mask by percentage of size of horizontal_x or horizontal_y, whichever is greater
+
+        in_place : boolean, default True
+            If `True` masks the current data, if `False` returns a new `ImageData` object.
+
+
+        Returns
+        -------
+        ImageData
+            If `in_place = False` returns a new ImageData object with the masked data
+
+        """
+        ig = self.geometry
+
+        # grid
+        y_range = (ig.voxel_num_y-1)/2
+        x_range = (ig.voxel_num_x-1)/2
+
+        Y, X = numpy.ogrid[-y_range:y_range+1,-x_range:x_range+1]
+
+        # use centre from geometry in units distance to account for aspect ratio of pixels
+        dist_from_center = numpy.sqrt((X*ig.voxel_size_x+ ig.center_x)**2 + (Y*ig.voxel_size_y+ig.center_y)**2)
+
+        size_x = ig.voxel_num_x * ig.voxel_size_x
+        size_y = ig.voxel_num_y * ig.voxel_size_y
+
+        if size_x > size_y:
+            radius_applied =radius * size_x/2
+        else:
+            radius_applied =radius * size_y/2
+
+        # approximate the voxel as a circle and get the radius
+        # ie voxel area = 1, circle of area=1 has r = 0.56
+        r=((ig.voxel_size_x * ig.voxel_size_y )/numpy.pi)**(1/2)
+
+        # we have the voxel centre distance to mask. voxels with distance greater than |r| are fully inside or outside.
+        # values on the border region between -r and r are preserved
+        mask =(radius_applied-dist_from_center).clip(-r,r)
+
+        #  rescale to -pi/2->+pi/2
+        mask *= (0.5*numpy.pi)/r
+
+        # the sin of the linear distance gives us an approximation of area of the circle to include in the mask
+        numpy.sin(mask, out = mask)
+
+        # rescale the data 0 - 1
+        mask = 0.5 + mask * 0.5
+
+        # reorder dataset so 'horizontal_y' and 'horizontal_x' are the final dimensions
+        labels_orig = self.dimension_labels
+        labels = list(labels_orig)
+
+        labels.remove('horizontal_y')
+        labels.remove('horizontal_x')
+        labels.append('horizontal_y')
+        labels.append('horizontal_x')
+
+
+        if in_place == True:
+            self.reorder(labels)
+            numpy.multiply(self.array, mask, out=self.array)
+            self.reorder(labels_orig)
+
+        else:
+            image_data_out = self.copy()
+            image_data_out.reorder(labels)
+            numpy.multiply(image_data_out.array, mask, out=image_data_out.array)
+            image_data_out.reorder(labels_orig)
+
+            return image_data_out
+
+
+class VectorData(DataContainer):
```

Comment:
Can `VectorData` and `VectorGeometry` be pulled out too?

Reply:
done (f712ea4)